### ISSUE TITLE
Functions prefixed with "pthread_t" get wrong scope

### ISSUE THREAD

**Issue by alexr00** [jeff-hykin/cpp-textmate-grammar#452]
I'm seeing the `pthread_t` part of` pthread_threadid_np();` get scoped as `support.type.built-in.posix-reserved.pthread.cpp`

From @foreignmeloman in https://github.com/microsoft/vscode/issues/91529

Seems like VS Code does not highlight correctly function calls which have "pthread_t" prefix in the name:

Examples:
```C++
pthread_threadid_np();
pthread_tfoo()
```

Screenshot:
![image](https://user-images.githubusercontent.com/11769352/75319975-1981fd00-5887-11ea-9001-a398480f8523.png)

**Comment by foreignmeloman** [comment]
I think this has something to do with `pthread_types` key of file [syntaxes/cpp.tmLanguage.json](https://github.com/jeff-hykin/cpp-textmate-grammar/blob/master/syntaxes/cpp.tmLanguage.json).
So the issue includes these prefixes:
```
pthread_attr_t
pthread_cond_t
pthread_condattr_t
pthread_mutex_t
pthread_mutexattr_t
pthread_once_t
pthread_rwlock_t
pthread_rwlockattr_t
pthread_t
```

but not the last one in the regex, which is `pthread_key_t`. Do we miss the grouping parenthesis here?

**Comment by jeff-hykin** [comment]
Fix pushed with `v1.14.24`
